packages/platform-browser/src/internal/geolocation.ts

```diff
@@ -0,0 +1,35 @@
+import * as Chunk from "effect/Chunk"
+import * as Context from "effect/Context"
+import * as Effect from "effect/Effect"
+import * as Layer from "effect/Layer"
+import * as Option from "effect/Option"
+import * as Stream from "effect/Stream"
+import type { Geolocation } from "../Geolocation.js"
+
+/** @internal */
+export const tag = Context.GenericTag<Geolocation>("@effect/platform-browser/Geolocation")
+
+/** @internal */
+export const layer = Layer.succeed(
+  tag,
+  {
+    getCurrentPosition: (options?: PositionOptions) =>
+      Effect.async<GeolocationPosition, GeolocationPositionError>((resume) =>
+        navigator.geolocation.getCurrentPosition(
+          (position) => resume(Effect.succeed(position)),
+          (error) => resume(Effect.fail(error)),
+          options
+        )
+      ),
+    watchPosition: (options?: PositionOptions) =>
+      Stream.async<GeolocationPosition, GeolocationPositionError>((emit) => {
+        const handlerId = navigator.geolocation.watchPosition(
+          (position) => emit(Effect.succeed(Chunk.of(position))),
+          (error) => emit(Effect.fail(Option.some(error))),
```

Comment:
If this errors, is the watcher also cancelled?

Reply:
I can't say for sure - but I think unsubscribing happens automatically because an error callback is optional

Reply:
after a little research, I realized that it is worth unsubscribing only after an `PermissionDeniedError`
the other two errors do not affect the receipt of events with coordinates